experimental/FTheoryTools/src/auxiliary.jl

```diff
@@ -525,3 +525,51 @@ true
 function _martins_desired_blowup(v::NormalToricVarietyType, I::MPolyIdeal; coordinate_name::Union{String, Nothing} = nothing)
   return _martins_desired_blowup(v, ideal_sheaf(v, I))
 end
+
+
+
+###########################################################################
+# 10: Apply a function to the innermost nested structure
+###########################################################################
+
+function deepmap(f, x)
+    if x isa AbstractArray
+        return map(e -> deepmap(f, e), x)
+    else
+        return f(x)
+    end
+end
+
+
+
+###########################################################################
+# 11: Macro for function generation
+###########################################################################
+
+macro define_model_attribute_getter(arg_expr, doc_example="")
+  if !(arg_expr isa Expr && arg_expr.head == :tuple && length(arg_expr.args) == 2)
+    error("Expected input like: (function_name, ReturnType)")
+  end
+
+  fname_expr = arg_expr.args[1]
+  rettype_expr = arg_expr.args[2]
+  fname = fname_expr isa Symbol ? fname_expr : eval(fname_expr)
```

Comment:
```suggestion
  fname = fname_expr isa Symbol ? fname_expr : error("function_name is not a symbol")
```
Every eval is potentially dangerous and should thus be avoided. But from the places you use this macro, none of them should need the eval

Reply:
Done.